rascal2/dialogs/settings_dialog.py

```diff
@@ -0,0 +1,116 @@
+from typing import Callable
+
+from PyQt6 import QtCore, QtWidgets
+
+from rascal2.core.settings import Settings, SettingsGroups
+from rascal2.widgets.inputs import ValidatedInputWidget
+
+
+class SettingsDialog(QtWidgets.QDialog):
+    def __init__(self, parent):
+        """
+        Dialog to adjust RasCAL-2 settings.
+
+        Parameters
+        ----------
+        parent : MainWindowView
+            The view of the RasCAL-2 GUI
+        """
+        super().__init__(parent)
+
+        self.setModal(True)
+        self.setMinimumWidth(400)
+
+        self.settings = parent.settings.copy()
+        self.reset_dialog = None
+
+        tab_widget = QtWidgets.QTabWidget()
+        tab_widget.addTab(SettingsTab(self, SettingsGroups.General), SettingsGroups.General)
+        tab_widget.addTab(SettingsTab(self, SettingsGroups.Logging), SettingsGroups.Logging)
+
+        self.reset_button = QtWidgets.QPushButton("Reset to Defaults", self)
+        self.reset_button.clicked.connect(self.reset_default_settings)
+        self.accept_button = QtWidgets.QPushButton("OK", self)
+        self.accept_button.clicked.connect(self.update_settings)
+        self.cancel_button = QtWidgets.QPushButton("Cancel", self)
+        self.cancel_button.clicked.connect(self.reject)
+
+        button_layout = QtWidgets.QHBoxLayout()
+        button_layout.addWidget(self.reset_button)
+        button_layout.addStretch(1)
+        button_layout.addWidget(self.accept_button)
+        button_layout.addWidget(self.cancel_button)
+
+        main_layout = QtWidgets.QVBoxLayout()
+        main_layout.addWidget(tab_widget)
+        main_layout.addLayout(button_layout)
+        self.setLayout(main_layout)
+        self.setWindowTitle("Settings")
+
+    def update_settings(self) -> None:
+        """Accept the changed settings"""
+        self.parent().settings = self.settings.copy()
+        self.parent().settings.save(self.parent().save_path)
+        self.accept()
+
+    def reset_default_settings(self) -> None:
+        """Reset the settings to the global defaults"""
```

Comment:
The _setting.json_ file needs to be removed after a reset. maybe add a clear method in setting to delete it